src/pages/patientView/trialMatch/TrialMatchTableUtils.ts

```diff
@@ -0,0 +1,129 @@
+            } else {
+                positiveCancerTypes.push(item);
+            }
+        });
+        const clinicalGroupMatch: IClinicalGroupMatch = {
+            trialAgeNumerical: age,
+            trialOncotreePrimaryDiagnosis: {
+                positive: positiveCancerTypes,
+                negative: negativeCancerTypes
+            },
+            matches: [],
+            notMatches: []
+        };
+        groupTrialMatchesByGenomicAlteration(matchesGroupedByAge[age], clinicalGroupMatch);
+        matchedTrial.priority += setTrialMatchPriority(clinicalGroupMatch);
+        armMatch.matches.push(clinicalGroupMatch);
+    });
+}
+
+export function groupTrialMatchesByGenomicAlteration(ageGroup: ITrialMatch[], clinicalGroupMatch: IClinicalGroupMatch) {
+    const matchesGroupedByGenomicAlteration = _.groupBy(ageGroup, (trial: ITrialMatch) => trial.genomicAlteration);
+    _.forEach(matchesGroupedByGenomicAlteration, (genomicAlterationGroup, genomicAlteration) => {
+        const genomicGroupMatch = groupTrialMatchesByPatientGenomic(genomicAlterationGroup, genomicAlteration);
+        // If a genomic alteration(i.e. BRAF V600E) contains a "!",
+        // it means this trial cannot be used for the genomic alteration, which is a "NOT" match.
+        if(genomicAlteration.includes('!')) {
+            clinicalGroupMatch.notMatches.push(genomicGroupMatch);
+        } else {
+            clinicalGroupMatch.matches.push(genomicGroupMatch);
+        }
+    });
+}
+
+export function groupTrialMatchesByPatientGenomic(genomicAlterationGroup: ITrialMatch[], genomicAlteration: string): IGenomicGroupMatch {
+    const matchesGroupedByPatientGenomic = _.groupBy(genomicAlterationGroup, (trial: ITrialMatch) => trial.patientGenomic);
+    const genomicGroupMatch: IGenomicGroupMatch = {
+        genomicAlteration: genomicAlteration,
+        matches: []
+    };
+    _.forEach(matchesGroupedByPatientGenomic , (patientGenomicGroup: ITrialMatch[]) => {
+        const genomicMatch: IGenomicMatch = {
+            trueHugoSymbol: patientGenomicGroup[0].trueHugoSymbol,
+            trueProteinChange: patientGenomicGroup[0].trueProteinChange,
+            sampleIds: _.uniq(_.map(patientGenomicGroup, (trial: ITrialMatch) => trial.sampleId))
+        };
+        genomicGroupMatch.matches.push(genomicMatch);
+    });
+    return genomicGroupMatch;
+}
+
+export function setTrialMatchPriority(clinicalGroupMatch: IClinicalGroupMatch): number {
+    if (clinicalGroupMatch.notMatches.length > 0) {
+        if (clinicalGroupMatch.matches.length === 0) {
+            return 2;
+        }
+        return 1;
+    }
+    return 0;
+}
+
+export function getDrugsAndExcludeControlArms(armDescription: string, arms: IArm[], armMatch: IArmMatch): boolean {
+    const hiddenArmTypes = ['Control Arm', 'Placebo Arm'];
+    if ( armDescription !== '' ) { // match for specific arm
+        const matchedArm: IArm = _.find( arms, { 'arm_description': armDescription } )!;
```

Comment:
shouldn't use lodash shortcut (not typesafe)

Reply:
Changed to (arm) => arm.description === armDescription